### PR TITLE
Task completed: GP-291. ObserverArea and Behavior

### PR DESCRIPTION
**Pull request by Kjusok**
Implemented behavior: when there is no movement, the observer area moves from right to left

### CODE REVIEW

File: Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
 using UnityEngine;
 
 namespace Gameplay.AI
 {
     public class ObserveArea : MonoBehaviour
     {
-        public event Action TargetsChanged;
+        private const float MovementThreshold = 0.01f;
+        
+        [SerializeField][Range(0,360)] private int _angel = 30;
```
Comment (frnzee) [inline Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs]: rename to _angle

Comment (Kjusok) [reply Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs]: Done

File: Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs [hunk 2]
```diff
@@ -26,9 +35,53 @@ private void OnTriggerExit(Collider other)
...
+        private void RotationLeft()
+        {
+            _rotationTween = transform.DOLocalRotate(new Vector3(0f, -_angle, 0f), _rotationSpeed, RotateMode.Fast)
+                .OnComplete(RotationRight);
+        }
+        
+        private void KillRotationTween()
+        {
+            _rotationTween.Kill();
+            transform.rotation = new Quaternion();
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs]: Pls use Quaternion.identity

File: Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs [hunk 3]
```diff
@@ -26,9 +35,53 @@ private void OnTriggerExit(Collider other)
...
+        }
+
+        private void Update()
+        {
+            _isMoving = Mathf.Abs(transform.position.magnitude - _lastPosition.magnitude) > MovementThreshold ;
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs]: Pls change _isMoving var from private field to local var. ANd send this var to ProcessRotation() method

Comment (ega-ega) [inline Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs]: Pls use for calculate distance like this: Vector3.Distance(currentPosition, previousPosition)

File: Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs [hunk 4]
```diff
@@ -26,9 +35,53 @@ private void OnTriggerExit(Collider other)
...
+            _lastPosition = transform.position;
+
+            ProcessRotation();
+        }
+
+        private void ProcessRotation()
+        {
+            if (_isMoving && _rotationTween.IsActive())
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs]: I think its mistake in expresion. I suggest: !_isMoving && _rotationTween.IsActive() => KillRotation

**Review by ega-ega — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs (hunk 2 above)
Comment (Kjusok) [reply Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs]: Done

File: Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs (hunk 3 above)
Comment (Kjusok) [reply Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs]: Done

Comment (Kjusok) [reply Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs]: Done

File: Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs (hunk 4 above)
Comment (Kjusok) [reply Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs]: It's not a mistake. The logic is this: when _isMoving = true, the object should not rotate.

**Review by ega-ega — APPROVED** [review]

**Review by frnzee — APPROVED** [review]

**Pull request merged**